Source/VCGeneration/Split.cs

```diff
@@ -1558,5 +1558,10 @@ public void ReleaseChecker()
           checker = null;
         }
       }
-    }
+
+      public void Finish(VCResult result) {
+        Checker.TheoremProver.Close();
```

Comment:
Why would be close here? Shouldn't the theoremProver of the checker be reused?

Reply:
Good question, let me tell you why this is needed in my case, and so we can discuss alternatives.
When I ran the boogie tests on Windows, it kept crashing 90% of the tests on `GetProverLog.cs` on line 35 because it's trying to read the log files and these logs files were being written by the theorem prover and never closed (`StreamWriter`). So, somewhere I need to close these log files and this was never done explicitely except in the public `Close()` method of the theorem prover which was not call.
Any suggestions to do otherwise?

Reply:
The trick is to Dispose the ExecutionEngine before reading the log files, like it is done in Dafny [here](https://github.com/dafny-lang/dafny/blob/master/Source/DafnyPipeline.Test/IntraMethodVerificationStability.cs#L217).

Reply:
That's a good trick ! It works !